File: a-first-element-finder.py
def find_element(arr, func) :
    for item in arr:
        if func(item):
            return item
    return None

def find_element_pythonic(arr, func) :
    return next((item for item in arr if func(item)), None)

#callback ex

def find_element1(arr, func) :
    for item in arr:
        if func(item):
            return item
    return None

def is_even(num):
    return num % 2 ==0

print(find_element1([1,3,8], is_even))

File: a-first-element-finder.js
function findElement(arr,func) {
    for (let i =0; i< arr.length; i++) {
        if (func(arr[i])) {
            return arr[i];
        }
    }
    return undefined;
}

//

function findElement1(arr, func) {
    return arr.find(func);
}

//

function isEven(num) {
    return num % 2 === 0;
}
fundElement([1,3,8], isEven);

findElement([1,3,8], num => num % 2 ===0);